### DISCUSSION TITLE
Does GHCP crawl through my code repository to suggest code?

### DISCUSSION THREAD

**Discussion by raghuram-jembu** [community/community#144903] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have a project written in Dotnet and stored code in my Azure Devops. and have a few files opened in the Visual studio. I have GHCP enterprise version. When i provide a prompt or context  to generate new code suggestion, does the GHCP would be able to crawl my Azure Devops repository and get influenced by all my code or it will search only in the public repository and suggest code?

**Comment by lucasthalless** [comment]
In GitHub's General Privacy Statement (here: https://docs.github.com/pt/site-policy/privacy-policies/github-general-privacy-statement) it is stated GitHub collects data to operate and improve services, but respects access permissions to repositories. GitHub Copilot does not automatically have access to private repositories unless the user has explicitly granted this permission. Copilot uses public code and the context of the code you have opened in the editor, and does not automatically track repositories like Azure DevOps.

The GitHub Copilot documentation (https://docs.github.com/en/copilot) clarifies that the plugin works based on files opened in your editor (e.g. VS Code or Visual Studio). Locally open source may influence suggestions, but there is no automatic scanning of externally stored code on platforms like Azure DevOps.
